ios/engine/KMEI/KeymanEngine/Classes/HTTPDownloadRequest.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  HTTPDownloadRequest.swift
+//  KeymanEngine
+//
+//  Created by Gabriel Wong on 2017-09-15.
+//  Copyright © 2017 SIL International. All rights reserved.
+//
+
+@objc public enum DownloadType: Int {
+  case downloadFile
+  case downloadCachedData
+}
+
+public class HTTPDownloadRequest: NSObject {
+  @objc public let url: URL
+  @objc public let typeCode: DownloadType
+  // TODO: Make values of this dict properties of the class
+  @objc public var userInfo: [String: Any]
+  var task: URLSessionTask?
+  @objc public var destinationFile: String?
+  @objc public var rawResponseData: Data?
+  @objc public var tag: Int = 0
+
+  @objc public init(url: URL, downloadType type: DownloadType, userInfo info: [String: Any]) {
+    self.url = url
+    typeCode = type
+    userInfo = info
+    super.init()
+  }
+
+  @objc public convenience init(url: URL) {
+    self.init(url: url, downloadType: DownloadType.downloadFile, userInfo: [:])
+  }
+
+  @objc public convenience init(url: URL, userInfo info: [String: Any]) {
+    self.init(url: url, downloadType: DownloadType.downloadFile, userInfo: info)
+  }
+
+  @objc public convenience init(url: URL, downloadType type: DownloadType) {
+    self.init(url: url, downloadType: type, userInfo: [:])
+  }
+
+  var responseStatusCode: Int? {
+    if let response = task?.response as? HTTPURLResponse {
+      return response.statusCode
+    }
```

Comment:
What happened to the 'else' condition from the Obj-C version of the file?  Is `task.response` guaranteed to be of the appropriate type now?

Also, there's that "TODO:  Remove" right after this section...

Reply:
The ObjC version threw an error if the response was not an `HTTPURLResponse` but now we just return `nil` in the case where there's no error code. I also recall the only caller is `KMManager.downloadRequestFinished:` which assumed a status code would be present anyway.

And yes that `@objc` property should be removed now that we don't need to be called by ObjC. It was only needed since `Int?` cannot be bridged.